code/ATMOSPHERICS/components/binary_devices/pump.dm

```diff
@@ -97,14 +97,24 @@ Thus, the two variables affect pump operation are set in New():
 
 	return 1
 
-/obj/machinery/atmospherics/binary/pump/interact(mob/user as mob)
-	var/dat = {"<b>Power: </b><a href='?src=\ref[src];power=1'>[on?"On":"Off"]</a><br>
-				<b>Desirable output pressure: </b>
-				[round(target_pressure,0.1)]kPa | <a href='?src=\ref[src];set_press=1'>Change</a>
-				"}
+/obj/machinery/atmospherics/binary/pump/ui_interact(mob/user, ui_key = "main")
+	if(stat & (BROKEN|NOPOWER))
+		return
+
+	var/data = list()
+
+	data["on"] = on
+	data["pressure_set"] = round(target_pressure*100) //Nano UI can't handle rounded non-integers, apparently.
+	data["max_pressure"] = MAX_OUTPUT_PRESSURE
 
-	user << browse("<HEAD><TITLE>[src.name] control</TITLE></HEAD><TT>[dat]</TT>", "window=atmo_pump")
-	onclose(user, "atmo_pump")
+	var/datum/nanoui/ui = SSnano.get_open_ui(user, src, ui_key)

```

Comment:
That's probably a really good idea.
If I ever bother with a part two then consider it part of it. (But considering how much work it was to get this far...)